Submitted By: Ken Moffat <ken at linuxfromscratch dot org>
Date: 2016-05-07
Initial Package Version: 5.6.0
Upstream Status: Applied
Origin: Upstream
Description:  Fixes build failure with nss-3.23.

From: Kai Koehne <kai.koehne@theqtcompany.com>
Date: Thu, 24 Mar 2016 13:55:28 +0100
Subject: [PATCH 1/1] Use system NSS only for certificate handling

Compiling against NSS 3.23 fails with current Chromium. Also, with NSS
3.21 there are failures connecting to e.g. google.com.

Fix this by adapting the setup endorsed by upstream Chromium: BoringSSL
is always used for cryptography, and NSS only for certificate handlng.

0472123 Fix build against newer NSS

Task-number: QTBUG-51890
Change-Id: If8aaed9b9a09475c5ed0dfec64d31f45ce9670f5
Reviewed-by: Allan Sandfeld Jensen <allan.jensen@theqtcompany.com>

diff -Naur a/qtwebengine/src/core/config/linux.pri b/qtwebengine/src/core/config/linux.pri
--- a/qtwebengine/src/core/config/linux.pri	2016-03-03 16:02:28.000000000 +0000
+++ b/qtwebengine/src/core/config/linux.pri	2016-05-06 22:49:35.557327092 +0100
@@ -18,11 +18,14 @@
     use_gio=0 \
     use_gnome_keyring=0 \
     use_kerberos=0 \
-    use_pango=0
+    use_pango=0 \
+    use_openssl=1
 
-!use?(nss) {
+use?(nss) {
+    GYP_CONFIG += use_nss_certs=1 \
+        use_openssl_certs=0
+} else {
     GYP_CONFIG += use_nss_certs=0 \
-        use_openssl=1 \
         use_openssl_certs=1
 }
 
